pipelines.go

```diff
@@ -55,6 +56,60 @@ func (p *Pipeline) Lock() *Pipeline {
 	return p
 }
 
+func (p *Pipeline) ValidateRefIds() error {
+
+	var refidsSet = make(map[string]bool)
+	var requisitestagerefSet = make(map[string]bool)
+
+	//TODO: What should I return if stages do not exists? I think is valid to not have
```

Comment:
Yes, a Spinnaker pipeline can be valid without stages.

One behavior we might consider here is returning a "fallible" warning. That is, we could return something like a `WarningEmptyStage` error and handle that upstream. This way tools like pacrd or dinghy could optionally log a warning if no stages are provided, thereby cutting down the amount of debugging someone has to do.

Reply:
I will check on that

Reply:
I created a struct ValidationResult that will contain a slice of error and warnings, the code will be changed so if there are multiple errors this will be reflected also, like the test refIds_and_stageref_do_not_exists that will be added.

```
type ValidationResult struct {
	Errors   []error
	Warnings []string
}